File: 2363-merge-similar-items/2363-merge-similar-items.cpp
class Solution {
public:
    vector<vector<int>> mergeSimilarItems(vector<vector<int>>& items1, vector<vector<int>>& items2) {
        vector<vector<int>>vec;
        vector<vector<int>>ans;
        for(int i=0;i<items1.size();i++){
            vec.push_back(items1[i]);
        }
        for(int i=0;i<items2.size();i++){
            vec.push_back(items2[i]);
        }
        sort(vec.begin(),vec.end());
        int i = 0;
        while(i < vec.size()-1)
        {
            if(vec[i][0] == vec[i+1][0])
            {
                int sum = vec[i][1] + vec[i+1][1];
                ans.push_back({vec[i][0], sum});
                i+=2;                
            }
            else 
            {
                ans.push_back({vec[i][0], vec[i][1]});
                i++;
            }
        }
        if(i == vec.size()-1)
            ans.push_back({vec[i][0], vec[i][1]});   
        return ans;
    }
};

File: 2363-merge-similar-items/2363-merge-similar-items.java
class Solution {
    public List<List<Integer>> mergeSimilarItems(int[][] items1, int[][] items2) 
    {
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i = 0; i<items1.length; i++)
        {
            map.put(items1[i][0], items1[i][1]);
        }
        for(int i = 0; i<items2.length; i++)
        {
            if(map.containsKey(items2[i][0]))                
            {
                map.put(items2[i][0], map.get(items2[i][0]) + items2[i][1]);
            }
            else 
            {
                map.put(items2[i][0], items2[i][1]);
            }
        }
        List<List<Integer>> ans = new ArrayList<>();
        for(Map.Entry<Integer, Integer> m : map.entrySet())
        {
            ArrayList<Integer> temp = new ArrayList<>();
            temp.add(m.getKey());
            temp.add(m.getValue());
            ans.add(temp);
        }
        Collections.sort(ans, new Comparator<List<Integer>>(){
            @Override
            public int compare(List<Integer> a, List<Integer> b)
            {
                return a.get(0) < b.get(0) ? -1 : 1; 
            }
        });
        return ans;
    }
}